st2common/bin/migrations/v3.5/st2-migrate-db-dict-field-values

```diff
@@ -0,0 +1,306 @@
+        print("")
+
+    for task_execution_id in task_execution_ids or []:
+        task_execution_db = TaskExecution.get_by_id(task_execution_id)
+        print("Migrating TaskExecutionDB with id %s" % (task_execution_db.id))
+
+        task_execution_db._mark_as_changed("task_spec")
+        task_execution_db._mark_as_changed("context")
+        task_execution_db._mark_as_changed("result")
+
+        task_execution_db.save()
+        print("TaskExecutionDB with id %s has been migrated" % (task_execution_db.id))
+        print("")
+
+
+def migrate_triggers() -> None:
+    print("Migrating trigger objects")
+
+    trigger_instance_dbs = TriggerInstanceDB.objects(
+        __raw__={
+            "payload": {
+                "$not": {
+                    "$type": "binData",
+                },
+            },
+            "status": {
+                "$in": TRIGGER_INSTANCE_COMPLETED_STATES,
+            },
+        }
+    ).only("id")
+    trigger_instance_ids = [
+        str(trigger_instance_db.id) for trigger_instance_db in trigger_instance_dbs
+    ]
+
+    if not trigger_instance_ids:
+        print("Found no TriggerInstanceDB objects to migrate.")
+        return None
+
+    print("Will migrate %s TriggerInstanceDB objects" % (len(trigger_instance_ids)))
+
+    for trigger_instance_id in trigger_instance_ids or []:
+        trigger_instance_db = TriggerInstance.get_by_id(trigger_instance_id)
+        print("Migrating TriggerInstanceDB with id %s" % (trigger_instance_db.id))
+
+        trigger_instance_db._mark_as_changed("payload")
+
+        trigger_instance_db.save()
+        print(
+            "TriggerInstanceDB with id %s has been migrated" % (trigger_instance_db.id)
+        )
+        print("")
+
+
+def migrate_objects(display_prompt: bool = True) -> None:
+    print("StackStorm v3.5 database field data migration script\n")
+
+    if display_prompt:
+        input(
+            "You are stronly recommended to create database backup before running the "
+            "migration script.\n\nDepending on the number of the objects in the database, "
+            "migration may take multiple hours or more. You are recommended to start the the "
+            "script in a screen session or similar. \n\n"
+            "To proceed with the migration, press enter and to cancel it, press CTRL+C."
+        )
+        print("")
+
+    print("Migrating affected database objects to utilize new field type")
+    print("")
+
+    start_ts = int(time.time())
+    migrate_executions()
+    migrate_workflow_objects()
+    migrate_triggers()
+    end_ts = int(time.time())
+
+    duration = end_ts - start_ts
+
+    print("")
+    print(
+        "SUCCESS: All database objects migrated successfully (duration: %s seconds)."
+        % (duration)
+    )
+
+
+def main():
+    if "--yes" in sys.argv:
+        sys.argv.remove("--yes")
+        display_prompt = False
+    else:
+        display_prompt = True
+
+    config.parse_args()
+    db_setup()
```

Comment:
Is it intended that exceptions while estabilishing the db connection aren't handled?

My first idea was that exceptions should be handled but it may be ok without  explicit exception handling since any error would immediately be visible to the user. So I don't have a strong opinion on this topic but catching exceptions and printing a message like the one below when an exception during migrate_objects is raised is probably a bit more user-friendly.